src/planner/mod.rs

```diff
@@ -297,8 +297,13 @@ pub enum PlannerError {
     InstallSettings(#[from] InstallSettingsError),
     /// A MacOS (Darwin) plist related error
     #[error(transparent)]
+    #[cfg(target_os = "macos")]
```

Comment:
Is there any way we could not guard these error variants? Otherwise, downstream consumers (of which there are none, right now, to be fair...) will likewise have to guard their potential handling of these errors -- this could lead to "compiles on Linux but not on macOS" (or vice-versa) issues for them.

Not a particularly strong feeling, but I purposely didn't add a guard to `Wsl1` for that reason.

(We should also guard the `NixOs` variant on being Linux if we _do_ decide we want to guard these errors.)

Reply:
Hmmm... this is marked `#[non_exhaustive]` so it's my impression if someone wanted to handle these errors they'd want to handle it in platform specific code, or capture it in the wildcard.

I'm not particularly bothered if we remove the platform specific stuff here, so let's do that. :)